userspace/libscap/linux/scap_procs.c

```diff
@@ -585,15 +585,15 @@ static int32_t scap_proc_add_from_proc(struct scap_linux_platform* linux_platfor
 		f = fopen(filename, "r");
 		if(f == NULL)
 		{
-			return SCAP_SUCCESS;
+			return scap_errprintf(error, errno, "can't find valid proc dir in %s", dir_name);
```

Comment:
In the past, returning SCAP_SUCCESS was fair enough, since the actual result thread info allocation happened below and the caller could still check for a NULL out thread info. Now, we can’t rely on the NULL result, since the caller is responsible of allocating it. As a consequence, for invalid thread IDs we always returned bogus scap thread infos. So IMO this is easily fixable by returning SCAP_FAILURE there, since they are actual failure scenarios.